hub/core/meta/tensor_meta.py

```diff
@@ -69,16 +71,61 @@ def create(
     def load(key: str, storage: StorageProvider):
         return TensorMeta(get_tensor_meta_key(key), storage)
 
-    def update_tensor_meta_with_array(self, array: np.ndarray, batched=False):
+    def check_batch_is_compatible(self, array: np.ndarray):
+        """Check if this `tensor_meta` is compatible with `array`. The provided `array` is treated as a batch of samples.
+
+        Note:
+            If no samples exist in the tensor this `tensor_meta` corresponds with, `len(array.shape)` is not checked.
+
+        Args:
+            array (np.ndarray): Batched array to check compatibility with.
+
+        Raises:
+            TensorMetaMismatchError: Dtype for array must be equal to this meta.
+            TensorInvalidSampleShapeError: If a sample already exists, `len(array.shape)` has to be consistent for all arrays.
+        """
+
+        if self.dtype != array.dtype.name:
```

Comment:
This was causing errors for me as the default htype was applied.
Changed this line to: 
`if self.length > 0 and self.dtype != array.dtype.name:`

So once dtype was set then it made sense to check this. I mentioned this in a different comment as well.

Reply:
@McCrearyD I made the change as we are near a deadline, can u please review this change that I made to your PR.

Reply:
i updated this to work with elastic dtypes